addons/website/tests/test_res_users.py

```diff
@@ -64,3 +66,30 @@ def create_user_pou():
         # rollback in retrying().
         with TestCase.assertRaises(self, ValidationError), mute_logger('odoo.sql_db'):
             retrying(create_user_pou, env)
+
+    def _create_user_via_website(self, website, login):
+        # We need a fake request to _signup_create_user.
+        with MockRequest(self.env, website=website):
+            return self.env['res.users'].with_context(website_id=website.id)._signup_create_user({
+                'name': login,
+                'login': login,
+            })
+
+    def test_portal_become_internal(self):
+        # This tests if the website_id is correctly removed when a portal user
+        # becomes an internal user.
+        website = self.env['website'].create({'name': "Awesome Website"})
+        # Permit uninvited signup for portal users.
+        website.auth_signup_uninvited = 'b2c'
+        # Disable cross-website for portal users.
+        website.specific_user_account = True
+        user = self._create_user_via_website(website, 'Portal_User')
+        self.assertEqual(user.website_id, website)
+        self.assertEqual(user._is_portal(), True)
+        self.assertTrue("Portal" in user.groups_id.mapped('name'))
+        internal_group = self.env['res.groups'].search([('name', '=', "Internal User")])
+        portal_group = self.env['res.groups'].search([('name', '=', "Portal")])
```

Comment:
use self.env.ref
```suggestion
        portal_group = self.env.ref('base.group_portal')
```

Reply:
I didn't find `base.group_internal` :disappointed: What is the proper way to get the internal group ?

Reply:
I don't think trying to find `base.group_internal` randomly was the right way to find the group :wink: Well, it was worth a shot but otherwise:

You are searching for a group whose name is "Internal User" => Find where that one is defined, you'll find:
```xml
<record model="res.groups" id="group_user">
    <field name="name">Internal User</field>
</record>
``` 
=> So the xml id is `base.group_user` :shrug: